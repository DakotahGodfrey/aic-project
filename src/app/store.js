import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
