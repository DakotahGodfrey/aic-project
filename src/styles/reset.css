*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
