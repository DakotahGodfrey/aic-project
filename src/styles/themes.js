const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
