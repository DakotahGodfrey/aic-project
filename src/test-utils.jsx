import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
