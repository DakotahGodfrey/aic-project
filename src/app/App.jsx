import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
