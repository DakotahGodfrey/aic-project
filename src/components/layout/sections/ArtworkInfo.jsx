import styled from 'styled-components';
import { getFromTheme } from 'styles/themes';
import TitleBlock from 'components/layout/headers/TitleBlock';
import AccordionCard from 'components/layout/cards/AccordionCard';
const DescriptionList = styled.dl`
  display: grid;
  grid-template-columns: max-content auto;
  font-size: 1.7rem;
  gap: 1.5rem 0;
  margin-top: 2rem;
  padding: 0.5rem;
  color: ${getFromTheme('textColor')};
  dt {
    padding: 1rem 2rem 0 0;
    font-weight: 900;
    border-top: 1px solid #b3b3b3;
  }
  dd {
    padding-top: 1rem;
    border-top: 1px solid #b3b3b3;
  }
`;

const ArtworkInfo = ({ artworkData, columns = 3 }) => {
  const {
    title,
    date_display,
    artist_display,
    artist_title,
    place_of_origin,
    medium_display,
    dimensions,
    credit_line,
    main_reference_number,
    publication_history,
    exhibition_history,
    provenance_text,
  } = artworkData;
  return (
    <>
      <div>
        <div>
          {title && (
            <TitleBlock>
              <h1>{title}</h1>
              <p>{date_display}</p>
              <p className='artist-display'>{artist_display}</p>
            </TitleBlock>
          )}
          <DescriptionList>
            <dt>Artist</dt>
            <dd>{artist_title && artist_title}</dd>
            <dt>Origin</dt>
            <dd>{place_of_origin && place_of_origin}</dd>
            <dt>Medium</dt>
            <dd>{medium_display && medium_display}</dd>
            <dt>Dimensions</dt>
            <dd>{dimensions && dimensions}</dd>
            <dt>Credit Line</dt>
            <dd>{credit_line && credit_line}</dd>
            <dt>Reference Number #</dt>
            <dd>{main_reference_number && main_reference_number}</dd>
          </DescriptionList>
        </div>
        {publication_history && publication_history !== null && (
          <AccordionCard
            heading={'Publication History'}
            body={publication_history.split('\n')}
            isArray
          />
        )}
        {exhibition_history && exhibition_history !== null && (
          <AccordionCard
            heading={'Exhibition History'}
            body={exhibition_history.split('\n')}
            isArray
          />
        )}
        {provenance_text && provenance_text !== null && (
          <AccordionCard heading={'Provenance'} body={provenance_text} />
        )}
      </div>
    </>
  );
};

export default ArtworkInfo;
