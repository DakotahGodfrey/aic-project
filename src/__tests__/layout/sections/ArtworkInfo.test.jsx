import { store } from 'app/store';
import ArtworkInfo from 'components/layout/sections/ArtworkInfo';
import { defaultTheme } from 'styles/themes';
import { screen, render } from 'test-utils';

const artworkData = {
  title: 'Artwork Title',
  artist_display: 'Artist Name',
  date_display: 'Date',
  artist_title: 'Artist Title',
  place_of_origin: 'Place of Origin',
  medium_display: 'Medium display',
  dimensions: 'Dimensions text',
  credit_line: 'credit line text',
  main_reference_number: 'main reference number 34',
  publication_history: 'publication history',
  exhibition_history: 'exhibition history',
  provenance_text: 'provenance text',
};

describe('<ArtworkInfo />', () => {
  it.skip('throws an error if no artworkData is provided', () => {
    expect(() => {
      render(<ArtworkInfo />, { theme: defaultTheme, store: store });
    }).toThrow();
  });
  it('matches the snapshot', () => {
    const { container } = render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(container).toMatchSnapshot();
  });
  it('renders an h1 withthe artwork title', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByRole('heading')).toHaveTextContent(artworkData.title);
  });
  it('renders the date_display', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.date_display)).toBeTruthy();
  });
  it('renders the artist_display', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.artist_display)).toBeTruthy();
  });
  it('renders the artist title if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.artist_title)).toBeTruthy();
  });
  it('renders the place_of_origin if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.place_of_origin)).toBeTruthy();
  });
  it('renders the medium_display if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.medium_display)).toBeTruthy();
  });
  it('renders the dimensions  if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.dimensions)).toBeTruthy();
  });
  it('renders the credit_line if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.credit_line)).toBeTruthy();
  });
  it('renders the main_reference_number if it exists', () => {
    render(<ArtworkInfo artworkData={artworkData} />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getByText(artworkData.main_reference_number)).toBeTruthy();
  });
});
