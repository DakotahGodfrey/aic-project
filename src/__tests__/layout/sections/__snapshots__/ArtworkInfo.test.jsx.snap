// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtworkInfo /> matches the snapshot 1`] = `
<div>
  <div>
    <div>
      <div
        class="sc-bdnxRM cIIiTt"
      >
        <h1>
          Artwork Title
        </h1>
        <p>
          Date
        </p>
        <p
          class="artist-display"
        >
          Artist Name
        </p>
      </div>
      <dl
        class="sc-dlnjwi jIUoQh"
      >
        <dt>
          Artist
        </dt>
        <dd>
          Artist Title
        </dd>
        <dt>
          Origin
        </dt>
        <dd>
          Place of Origin
        </dd>
        <dt>
          Medium
        </dt>
        <dd>
          Medium display
        </dd>
        <dt>
          Dimensions
        </dt>
        <dd>
          Dimensions text
        </dd>
        <dt>
          Credit Line
        </dt>
        <dd>
          credit line text
        </dd>
        <dt>
          Reference Number #
        </dt>
        <dd>
          main reference number 34
        </dd>
      </dl>
    </div>
    <div
      class="sc-gtsrHT hdYhOt"
    >
      <button
        aria-expanded="false"
      >
        Publication History
        <span
          class="material-icons"
        >
          add
        </span>
      </button>
      <ul
        class="body"
      >
        <li>
          publication history
        </li>
      </ul>
    </div>
    <div
      class="sc-gtsrHT hdYhOt"
    >
      <button
        aria-expanded="false"
      >
        Exhibition History
        <span
          class="material-icons"
        >
          add
        </span>
      </button>
      <ul
        class="body"
      >
        <li>
          exhibition history
        </li>
      </ul>
    </div>
    <div
      class="sc-gtsrHT hdYhOt"
    >
      <button
        aria-expanded="false"
      >
        Provenance
        <span
          class="material-icons"
        >
          add
        </span>
      </button>
      <div
        class="body"
        role="dialog"
      >
        <p>
          provenance text
        </p>
      </div>
    </div>
  </div>
</div>
`;
